### PR TITLE
Refresh after delete file

### PR DESCRIPTION
**Pull request by PolarisXII**
- patched up 630-sus-out-current-workspacefile-implementation
- force refresh workspace selector after file/folder gets deleted

### CODE REVIEW

File: client/src/visualiser-debugger/Component/FileTree/FileSelector.tsx [hunk 1]
```diff
@@ -1,137 +1 @@
...
-            {file.name}
-          </SelectItem>
-        ))}
-      </Select>
-    </div>
-  );
-};
-
-export default FileSelector;
+// @ts-nocheck
```
Comment (Dammshine) [inline client/src/visualiser-debugger/Component/FileTree/FileSelector.tsx:1]: Please remove.

File: client/src/visualiser-debugger/Component/FileTree/WorkspaceSelector.tsx [hunk 2]
```diff
@@ -1,186 +0,0 @@
...
-      </div>
-      <FileSelector
-        onChangeProgramName={onChangeProgramName}
-        getCurrentWorkspaceName={getCurrentWorkspaceName}
-        retrieveWorkspace={retrieveWorkspace}
-      />
-    </Box>
-  );
-};
-export default WorkspaceSelector;
```
Comment (Dammshine) [inline client/src/visualiser-debugger/Component/FileTree/WorkspaceSelector.tsx:186]: please remove

File: client/src/visualiser-debugger/Component/FileTree/css/WorkspaceSelector.css [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+  margin-top: 0px;
+  margin-bottom: 4px;
+  height: 16px;
+}
+
+.icon-button {
+  margin: 0px;
+  padding: 0px;
+  width: 16px;
+}
```
Comment (Dammshine) [inline client/src/visualiser-debugger/Component/FileTree/css/WorkspaceSelector.css:15]: Not being used anymore, please remove.

File: client/src/visualiser-debugger/Component/FileTree/LeafFile.tsx [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+export interface LeafFileParam {
+  file: IFileFileNode;
+  depth: number;
+}
+
+const LeafFile = ({ file, depth }: LeafFileParam) => {
+  const { currFocusFilePath, setFocusFilePath, setFocusDirPath } = useUserFsStateStore();
+  const isHighlighted = currFocusFilePath === file.path;
+
+  useEffect(() => {}, [currFocusFilePath]);
```
Comment (Dammshine) [inline client/src/visualiser-debugger/Component/FileTree/LeafFile.tsx:16]: What's this line for? 
"useEffect(() => {}, [currFocusFilePath]);"

File: client/src/visualiser-debugger/Component/FileTree/Folder.tsx [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+  const currFocusDir = fileSystem.getDirFromPath(currFocusDirPath);
+
+  const expandFolder = () => {
+    if (!isExpanded) {
+      setFocusDirPath(folder.path);
+    }
+    setExpanded(!isExpanded);
+  };
+
+  // useEffect(() => {
```
Comment (Dammshine) [inline client/src/visualiser-debugger/Component/FileTree/Folder.tsx:27]: please remove the comments

File: client/src/visualiser-debugger/Component/FileTree/FileManager.tsx [hunk 6]
```diff
@@ -0,0 +1,187 @@
...
+    } else {
+      fileSystem.deleteFile(dirToDelete);
+    }
+    fileSystem.saveChanges();
+    forceRerender();
+    currFocus = "root"
+    return;
+  }
+
+  console.log(currFocus);
```
Comment (Dammshine) [inline client/src/visualiser-debugger/Component/FileTree/FileManager.tsx:98]: please remove "console.log()"

File: client/src/visualiser-debugger/Component/FileTree/FileManager.tsx [hunk 7]
```diff
@@ -0,0 +1,187 @@
...
+
+const WorkspaceSelector = () => {
+  let { fileSystem, currFocusDirPath, currFocusFilePath } = useUserFsStateStore.getState();
+  let currFocus = currFocusFilePath || currFocusDirPath;
+  
+  const [open, setOpen] = useState(false);
+  const [newItemName, setNewItemName] = useState('');
+  const [type, setType] = useState<IFileType | null>(null);
+  
+  const [, forceRerender] = useReducer(x => x + 1, 0);
```
Comment (Dammshine) [inline client/src/visualiser-debugger/Component/FileTree/FileManager.tsx:27]: Hacky but fine, we need this

File: debugger/Dockerfile [hunk 8]
```diff
@@ -2,6 +2,9 @@ FROM ubuntu
 
 ENV DEBIAN_FRONTEND noninteractive
 
+RUN rm -rf /var/lib/apt/lists/*
+RUN apt-get clean
+RUN apt-get update -o Acquire::CompressionTypes::Order::=gz
```
Comment (Dammshine) [inline debugger/Dockerfile:7]: Ha, why we need this lol

**Review by Dammshine — CHANGES_REQUESTED** [review]

File: debugger/Dockerfile (hunk 8 above)
Comment (PolarisXII) [reply debugger/Dockerfile:7]: no idea.....is this to do w/ deployment?

File: client/src/visualiser-debugger/Component/FileTree/WorkspaceSelector.tsx (hunk 2 above)
Comment (PolarisXII) [reply client/src/visualiser-debugger/Component/FileTree/WorkspaceSelector.tsx:186]: which part?

**Review by Dammshine — APPROVED** [review]

**Pull request merged**